tests/models/test_evaluation.py

```diff
@@ -101,43 +101,43 @@ def spark_session():
     session.stop()
 
 
+class EvaluationDatasetWithSavedConstructor(EvaluationDataset):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self._constructor_args = kwargs
```

Comment:
I think we can define:

```python
Dataset = namedtuple('Dataset', ['data', 'targets', 'name'])
```

and replace:

```python
xxx_yyy_dataset._constructor_args["data"]
xxx_yyy_dataset._constructor_args["targets"]
```

with:

```python
xxx_yyy_dataset.data
xxx_yyy_dataset.targets
```

Reply:
We can address this later in a separate PR.

Reply:
Can we revert changes related to `EvaluationDatasetWithSavedConstructor`?

Reply:
Yes. So just go back to using `EvaluationDataset` with manual setting of `_constructor_args`? And then use `namedtuple` later?

Reply:
Yes, or you can file a PR to use `namedtuple` first.